### PR TITLE
Add EntityPairClassification Pipeline, AutoClass & LUKE ONNX Support 

### PR DESCRIPTION
**Pull request by kayvane1**
# What does this PR do?

This PR started out in adding support for Luke in ONNX.

To not break existing AutoPatterns in [FeaturesManager](src/transformers/onnx/features.py), [AutoModelForEntityPairClassification](src/transformers/models/auto/modeling_auto.py) has also been added.

Additionally, a pipeline for [EntityPairClassification](src/transformers/pipelines/entity_pair_classification.py) has been added to make the task more supported overall by the library.

Note: A previous PR (https://github.com/huggingface/transformers/pull/16562) has been closed / not merged for LUKE ONNX support. I believe this PR addresses the remaining comments in that one.

All ONNX tests pass - happy to implement any additional comments for the Pipeline / Autoclass.

I have only implemented one of the additional Tasks `EntityPairClassification` - if this has been done to the appropriate standard, I can also implement it for the other two remaining Luke Heads which are not currently supported `Span Classification` &  `Entity Classification`

@NielsRogge - Worked on the original LUKE implementation
@lewtun & @michaelbenayoun  - Reviewed the previous PR

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?
Screenshot 2022-08-08 at 13 27 00

### CODE REVIEW

File: src/transformers/models/auto/modeling_auto.py [hunk 1]
```diff
@@ -642,6 +642,13 @@
     ]
 )
 
+MODEL_FOR_ENTITY_PAIR_CLASSIFICATION_MAPPING_NAMES = OrderedDict(
+    # Model for Entity Pair Classification
+    [
+        ("luke", "LukeForEntityPairClassification"),
```
[the remark this answers is not in the document]
Comment (kayvane1) [reply src/transformers/models/auto/modeling_auto.py:648]: Relationship Classification is a more common use case in building up Knowledge Graphs, but you're right that the class would need to be made more generic to deal with LUKE's entity aware head vs. a generic transformer head.

[the remark this answers is not in the document]
Comment (kayvane1) [reply src/transformers/models/auto/modeling_auto.py:648]: Also - the [tests](tests/onnx/test_onnx_v2.py) are written with the idea that everything is an autoclass - it looks for a from_config method as part of the test. So would either need to add a conditional in the tests for LUKE, or a from_config method in the Model implementation. What do you think is the cleanest approach?

**Comment by kayvane1** [comment]
Hi @NielsRogge - what next steps would you suggest?
Happy to make updates to the PR

**Review by lewtun — COMMENTED** [review]
Thanks for adding ONNX support for this tricky model @kayvane1 ! 

As you point out, I think we have two options to handle this export:

1. Use the `LukeForXxx` classes in `features.py` and add a special condition in the ONNX tests to bypass the `AutoModel` construction
2. Use your new `AutoModelForEntityPairClassification` class to preserve the structure of the ONNX tests

As @NielsRogge points out, LUKE is a special model, so we're not likely to see much need for `AutoModelForEntityPairClassification`. 

This suggests we should adopt option (1), but gently pinging @sgugger @patrickvonplaten for their view on whether it makes sense to include:

* A new `AutoModelForEntityPairClassification` for LUKE-like models (are there any others?)
* A `EntityPairClassificationPipeline`

**Comment by sgugger** [comment]
I don't think it makes sense to create an auto-map just for this model, and the pipeline can be done as a [custom pipeline with code on the Hub](https://huggingface.co/docs/transformers/add_new_pipeline#share-your-pipeline-on-the-hub). If/when there are more models associated to this task, we can revisit this approach of course.

**Comment by kayvane1** [comment]
Thanks @NielsRogge , @lewtun & @sgugger !

I'll update the PR by reverting the autoclass creation and bypass the AutoModel Constructors in `test_onnx_v2.py` & use the LukeForXxx classes in `features.py` directly.

**Comment by NielsRogge** [comment]
Hi @kayvane1, feel free to revive this PR :)

**Pull request closed without merging**